rfcs/0048-on-conflict-clause-in-create-table-statement.md

```diff
@@ -0,0 +1,121 @@
+
+- **Ignore**
+  The Ignore behavior make user can define a "append-only table with primary key" which maintain the primary constraint and also provide a append-only changes to downstream with good performance. This is good to express a table with an At-least-once Delivery connector where a same record can be replayed multiple times to break the primary key constraint.
+  Notice: the specific grammar for this has not designed(e.g. should we expose append-only table to user?) but the situation is the same.
+
+- **Partial Update**
+  The ON CONFLICT make users can do partial update on the table.
+
+  ```SQL
+    CREATE table t(k int primary key, int a, int b) 
+    ON CONFLICT DO UPDATE SET (a,b) = (
+      COALESCE(EXCLUDED.a, a),
+      COALESCE(EXCLUDED.b, b));
+  ```
+  
+  This feature will be useful when user use wide-table schema data model.
+
+- **Express Reduce Function or UDAF?**
+  In fact, ON CONFLICT Clause can make user be able to do more things. User can even construct a streaming Map-reduce job with that where the primary key define the map fields and ON_CONFLICT clause define the reduce logic.
+  For example.
+
+  ```SQL
+    CREATE table t(k int primary key, int cnt) 
+    ON CONFLICT DO UPDATE SET cnt = t.cnt + EXCLUDED.cnt
+
+    Insert INTO t VALUES (1, 2);
+    Insert INTO t VALUES (1, 3);
+
+    dev=# select * from t;
+     k | cnt 
+    ---+-----
+     1 |   5
+    (1 row)
+  ```
+
+  The most strange part is that the input parameter and return value of the "reduce function" must have the same schema. But it can be easily workaround with constant expressions.
+  
+### Design
+
+#### Syntax and semantics
+
+```SQL
+  CREATE TABLE [ IF NOT EXISTS ] table_name (
+    col_name data_type [ PRIMARY KEY ],
+    ...
+    [ PRIMARY KEY (col_name, ... ) ]
+  ) ON CONFLICT conflict_action;
+
+  where conflict_action can be one of:
+    DO NOTHING
+    DO OVERWRITE
+    DO UPDATE SET { column_name = expression |
+                    ( column_name [, ...] ) = [ ROW ] ( expression[, ...] )
+                  } [, ...]
+              [ WHERE condition ]
+```
+
+I will compare it with the PG's ON CONFLICT clause(<https://www.postgresql.org/docs/current/sql-insert.html>) in the following text.
+
+```SQL
+INSERT INTO table_name [ AS alias ] [ ( column_name [, ...] ) ]
+    [ ON CONFLICT [ conflict_target ] conflict_action ]
+
+where conflict_target can be one of:
+    ( { index_column_name | ( index_expression ) } [ COLLATE collation ] [ opclass ] [, ...] ) [ WHERE index_predicate ]
+    ON CONSTRAINT constraint_name
+
+and conflict_action is one of:
+    DO NOTHING
+    DO UPDATE SET { column_name = { expression | DEFAULT } |
+                    ( column_name [, ...] ) = [ ROW ] ( { expression | DEFAULT } [, ...] ) |
+                    ( column_name [, ...] ) = ( sub-SELECT )
+                  } [, ...]
+              [ WHERE condition ]
+```
+
+- **Defined when `CREATE TABLE` instead of `INSERT`**
```

Comment:
I think we can consider this as a config for the data from connector. i.e. What command the connector should use to write data - REPLACE or INSERT or anything else.